src/MongoDB.Driver.Core/MongoConnectionException.cs

```diff
@@ -78,6 +80,24 @@ public ConnectionId ConnectionId
             get { return _connectionId; }
         }
 
+        /// <summary>
+        /// Whether or not this exception contains a timeout exception.
+        /// </summary>
+        public bool ContainsTimeOutException
```

Comment:
This name doesn't sound quite right... it sounds like it's looking for a `TimeoutException`.

I recommend renaming to `ContainsSocketTimeoutException`

Reply:
Done :)